### ISSUE TITLE
Error parsing tsconfig.json file in monorepo

### ISSUE THREAD

**Issue by bcheung** [francescov1/mongoose-tsgen#159]
Hello! I believe ts-node needs to be upgraded to fix a tsconfig.json issue with monorepos.

I'm getting this error below when I try to run mtgen:
```shell
Generating mongoose typescript definitions... !
    Error: Error parsing your tsconfig.json file:
    ENOENT: no such file or directory, open
    '/path/to/monorepo/packages/data/@tsconfig/node20/
    tsconfig.json'
``` 

I have a monorepo setup where my tsconfig.json file extends a base tsconfig installed from a package:
```javascript
{
  "extends": "@tsconfig/node20/tsconfig.json",
...
```

Using a relative path works in my tsconfig.json:
```javascript
{
  "extends": "../../node_modules/@tsconfig/node20/tsconfig.json",
...
``` 

See a similar issue here was solved with upgrading ts-node to v10.9.2:
https://github.com/microsoft/TypeScript/issues/56847#issuecomment-1873956762

**Comment by francescov1** [comment]
Hey @bcheung thanks for flagging, will try to look into it this weekend

**Comment by francescov1** [comment]
@bcheung This error is actually happening in the mongoose-tsgen code, right [here](https://github.com/francescov1/mongoose-tsgen/blob/01c72eb4d3237c7e62a271e65ec504c220a7ee10/src/helpers/tsReader.ts#L419). This isn't trivial to fix, we're currently using the `extends` field to find other tsconfig.json files and load in their `paths` fields, which ensures TS path resolution works. But we would need to resolve aliases fully to find aliased paths. 

Could you provide a simple reproduction repo? Would help a lot with fixing

**Comment by orgads** [comment]
@francescov1 https://github.com/orgads/js-import/tree/tsconfig-node-modules

**Comment by francescov1** [comment]
Thanks @orgads will look this weekend

**Comment by francescov1** [comment]
Sorry the for delay, will try to get to this shortly

**Comment by francescov1** [comment]
@bcheung @orgads fix is live in v9.4.3 🚀 path resolving and tsconfig parsing is much more robust now, using the node `resolve` module  and not overwriting the user's `compilerOptions.module` field

**Comment by francescov1** [comment]
Noticed a bug with tsconfig parsing of certain other `compilerOptions.module` options (not with your use case but other use cases), so pushed a further fix in v9.4.4. Try that one out, it should work for you as well.
